src/main/java/org/opensearch/security/identity/SecurityTokenManager.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.security.identity;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import joptsimple.internal.Strings;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.greenrobot.eventbus.Subscribe;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.common.transport.TransportAddress;
+import org.opensearch.identity.Subject;
+import org.opensearch.identity.noop.NoopSubject;
+import org.opensearch.identity.tokens.AuthToken;
+import org.opensearch.identity.tokens.OnBehalfOfClaims;
+import org.opensearch.identity.tokens.TokenManager;
+import org.opensearch.security.authtoken.jwt.ExpiringBearerAuthToken;
+import org.opensearch.security.authtoken.jwt.JwtVendor;
+import org.opensearch.security.securityconf.ConfigModel;
+import org.opensearch.security.securityconf.DynamicConfigModel;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.user.User;
+import org.opensearch.security.user.UserService;
+import org.opensearch.threadpool.ThreadPool;
+
+/**
+ * This class is the Security Plugin's implementation of the TokenManager used by all Identity Plugins.
+ * It handles the issuance of both Service Account Tokens and On Behalf Of tokens.
+ */
+public class SecurityTokenManager implements TokenManager {
+    private static final Logger logger = LogManager.getLogger(SecurityTokenManager.class);
+
+    private final ClusterService cs;
+    private final ThreadPool threadPool;
+    private final UserService userService;
+
+    private JwtVendor jwtVendor = null;
+    private ConfigModel configModel = null;
+
+    public SecurityTokenManager(final ClusterService cs, final ThreadPool threadPool, final UserService userService) {
+        this.cs = cs;
+        this.threadPool = threadPool;
+        this.userService = userService;
+    }
+
+    @Subscribe
+    public void onConfigModelChanged(final ConfigModel configModel) {
+        this.configModel = configModel;
+    }
+
+    @Subscribe
+    public void onDynamicConfigModelChanged(final DynamicConfigModel dcm) {
+        final Settings oboSettings = dcm.getDynamicOnBehalfOfSettings();
+        final Boolean enabled = oboSettings.getAsBoolean("enabled", false);
+        if (enabled) {
+            jwtVendor = createJwtVendor(oboSettings);
+        } else {
+            jwtVendor = null;
+        }
+    }
+
+    /** For testing */
+    JwtVendor createJwtVendor(final Settings settings) {
```

Comment:
Should this not also set the vendor for the instance's attribute?

Reply:
The jwt vendor is only created when the dynamic config is triggered, so it starts null and until the lifecycle of the security plugin has published the dynamic config update in the process it should be unset.